highfidelity/hifi
# PAL: Editable Display Names & Immediate Update

This PR allows users to edit their Display Name from within the PAL via an editable TextInput field. As a bonus, display names will update immediately. Other users who have their PALs open need only to refresh the PAL to see your shiny new name. Other users who open the PAL after you change your name will see the changes immediately.

I'm a little nervous about some of the changes in here because they modify code that @howard-stearns created 😄. Hopefully I didn't totally break any behavior he introduced.

**Test Plan**
This test plan requires three clients running this PR: Clients A, B, and C. Run Sandbox from Client A; Client A will have admin privileges on the domain.
* Set the Display Name to blank on all three clients.
* Using all three clients, connect to Sandbox A
* Open the PAL on all three clients.
* Depending on the order that the clients joined, each client will get one of three display names from the AvatarMixer: "Anonymous", "Anonymous_1", or "Anonymous_2". Across all three clients, verify that each client is correctly identified by their anonymous display name.
* On Client A, using the editable text field at the top of the PAL, change the display name to "Testing". (Pressing enter or de-focusing the text field updates the name.)
* On Clients B and C, refresh the PAL. Verify that Client A can now be identified as "Testing".
* On Client B, change the display name to "Testing".
* On all three clients, refresh the PAL. Verify that Client B can now be identified as "Testing_1".
* On Client A, change the display name to "Test". Then, change the display name to "Testing".
* On all three clients, refresh the PAL. Verify that Client A can now be identified as "Testing_2".
* On Client C, change the display name to "Best QA". Disconnect from Sandbox A, then reconnect to Sandbox A. On all three clients, verify that Client C can be verified as "Best QA"
